src/compileModel.ts

```diff
@@ -51,6 +56,25 @@ export interface ModelFindByIdOptions extends ModelDatabaseExecutionOptions {
 	/** Array of projection properties */
 	projection?: string[];
 }
+export interface ModelIncrementOptions extends ModelDatabaseExecutionOptions {
+	/**
+	 * Number of retries to perform when record is locked
+	 * @defaultValue 5
+	 */
+	retry?: number;
+	/**
+	 * Delay between retries in seconds when record is locked
+	 * @defaultValue 1
+	 */
+	retryDelay?: number;
+}
+
+export type IncrementOperation<TSchema extends Schema | null> = TSchema extends Schema
+	? {
+			path: keyof FlattenDocument<TSchema, Schema | SchemaDefinition | SchemaTypeDefinitionNumber>;
```

Comment:
This should also have the `Extract` on it since the called method expects it.

Reply:
Done. added Extract here.